src/main/java/core/basesyntax/dao/figure/FigureDaoImpl.java

```diff
@@ -3,20 +3,21 @@
 import core.basesyntax.dao.AbstractDao;
 import core.basesyntax.model.figure.Figure;
 import java.util.List;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
 
-public class FigureDaoImpl<T extends Figure> extends AbstractDao implements FigureDao<T> {
+public class FigureDaoImpl<T extends Figure> extends AbstractDao<T> implements FigureDao<T> {
     public FigureDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
-    @Override
-    public T save(T figure) {
-        return null;
-    }
-
     @Override
     public List<T> findByColor(String color, Class<T> clazz) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM " + clazz.getSimpleName()
+                    + " f WHERE f.color LIKE :color", clazz)
```

Comment:
why LIKE?

Reply:
I'm used to be using LIKE with the Strings everywhere
Should I use it only with the matching patterns?